### PR TITLE
Implement a new scalar: MultiplicationOf (#864)

### PR DESCRIPTION
**Pull request by Iprogrammerr**
MultiplicationOf and DivisionOf for supporting more math operations as described in #864 .

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#973](https://github.com/yegor256/cactoos/issues/973) is now in scope, role is `REV`

File: src/main/java/org/cactoos/scalar/DivisionOf.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+package org.cactoos.scalar;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+import org.cactoos.Scalar;
+
+/**
+ * Division result of two numbers.
+ *
+ * <p>This class implements {@link Scalar}, which throws a checked
```
Comment (llorllale) [inline src/main/java/org/cactoos/scalar/DivisionOf.java]: @Iprogrammerr this javadoc does not apply

File: src/test/java/org/cactoos/scalar/DivisionOfTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.IsEqual;
+import org.junit.Test;
+
+/**
+ * Test case for {@link DivisionOf}.
+ *
+ * @since 0.49.2
+ * @checkstyle JavadocMethodCheck (500 lines)
```
Comment (llorllale) [inline src/test/java/org/cactoos/scalar/DivisionOfTest.java]: @Iprogrammerr let's please add javadocs to the test methods

File: src/test/java/org/cactoos/scalar/MultiplicationOfTest.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+import org.cactoos.iterable.IterableOf;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.IsEqual;
+import org.junit.Test;
+
+/**
+ * Test case for {@link MultiplicationOf}.
+ *
+ * @since 0.49.2
+ * @checkstyle JavadocMethodCheck (500 lines)
```
Comment (llorllale) [inline src/test/java/org/cactoos/scalar/MultiplicationOfTest.java]: @Iprogrammerr let's please add javadocs to the test cases

File: src/main/java/org/cactoos/scalar/MultiplicationOf.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        this(new IterableOf<>(src));
+    }
+
+    /**
+     * Ctor.
+     * @param src The iterable
+     */
+    public MultiplicationOf(final Iterable<? extends Number> src) {
+        super(() -> new Folded<>(
+            BigDecimal.ONE,
```
Comment (llorllale) [inline src/main/java/org/cactoos/scalar/MultiplicationOf.java]: @Iprogrammerr so the result is `1` if no numbers are multiplied? I think we should force at least two arguments in the ctors with varargs. This one with `Iterable` should either be hidden or throw an `IllegalArgumentException`.

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#973` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#973](https://github.com/yegor256/cactoos/issues/973) is now out of scope

**Pull request merged**